Realm/Realm/Schema/ObjectSchema.cs

```diff
@@ -35,6 +35,16 @@ namespace Realms.Schema
     [DebuggerDisplay("Name = {Name}, Properties = {Count}")]
     public class ObjectSchema : IReadOnlyCollection<Property>
     {
+        // TODO andrea:
+        // 1 - move this to the right place
+        // 2- add documentation
+        public enum ObjectSchemaType : byte
+        {
+            TopLevel = 0,
+            Embedded = 1,
+            TopLevelAsymmetric = 2,
```

Comment:
If those are public, we should match the terminology we use for objects - i.e. `RealmObject`, `EmbeddedObject`, `AsymmetricObject`.